components/api/src/test/java/com/hotels/styx/api/HttpRequestTest.java

```diff
@@ -169,12 +166,6 @@ public void canModifyPreviouslyCreatedRequest() {
         assertThat(newRequest.headers(), hasItem(header("remove", "notanymore")));
     }
 
-    @Test
-    public void shouldDetermineIfRequestIsSecure() {
-        assertThat(get("https://hotels.com").build().isSecure(), is(true));
-        assertThat(get("http://hotels.com").build().isSecure(), is(false));
```

Comment:
I see an issue here. If we can still include the protocol when constructing the request, but then ignoring it, we are creating a misleading interface.

Reply:
These URLS are in absolute form. But the URL is usually specified in an "origin form" in which case the URL protocol scheme isn't present. See RFC 7320, chapter *5.3.  Request Target* for more details. Also see chapter *5.5.  Effective Request URI* for related information about http schemes.